File: solutions/0300_Longest_Increasing_Subsequence/dp.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        if not nums:
            return 0
        dp = [0] * len(nums)
        dp[0] = 1
        for i in range(1, len(nums)):
            k = 1
            for j in range(i):
                if nums[j] < nums[i]:
                    k = max(k, dp[j] + 1)
            dp[i] = k
        return max(dp)

File: solutions/0377_Combination_Sum_IV/dp.java
class Solution {
    public int combinationSum4(int[] nums, int target) {
        if (target == 0) {
            return 1;
        }
        int[] dp = new int[target + 1];
        dp[0] = 1;
        Arrays.sort(nums);
        for (int i = 1; i <= target; ++i) {
            int cnt = 0;
            for (int v : nums) {
                if (i - v >= 0 && dp[i - v] >= 0) cnt += dp[i - v];
            }
            dp[i] = cnt;
        }
        return dp[target];
    }
}
